lib/sensors/brakes.cpp

```diff
@@ -0,0 +1,38 @@
+#include "brakes.hpp"
+
+namespace hyped::sensors {
+
+std::optional<Brakes> Brakes::create(core::ILogger &logger,
+                                     std::shared_ptr<io::IGpio> gpio,
+                                     const std::uint8_t pin)
+{
+  const auto reader = gpio->getReader(pin, io::Edge::kNone);
+  if (!reader) {
+    logger.log(core::LogLevel::kFatal, "Failed to create Brakes instance");
+    return std::nullopt;
+  }
+  logger.log(core::LogLevel::kDebug, "Successfully created Brakes instance");
+  return Brakes(logger, *reader);
+}
+
+Brakes::Brakes(core::ILogger &logger, std::shared_ptr<io::IGpioReader> gpio_reader)
+    : gpio_reader_(gpio_reader),
+      logger_(logger),
+      pin_()
```

Comment:
Do you need this here? Either pass the pin into the constructor if it may need to be accessible in the future (i.e. if something else might want to know which pin it's using) or remove it from here and the header file